### DISCUSSION TITLE
How to setup external GPS module?

### DISCUSSION THREAD

**Discussion by drtorchwood** [cyberman54/ESP32-Paxcounter#800] [Q&A]
Hi, 

I have an external GPS module like this one https://www.aliexpress.com/item/1005001986419551.html and want to connect it to my TTGO Lora v2.1 (https://www.amazon.de/TTGO-ESP32-Paxcounter-LoRa32-Version-Bluetooth/dp/B07SFJ86FG). 
The part in the generic.h says:
```
#define HAS_GPS 1 // use on board GPS
#define GPS_SERIAL 9600, SERIAL_8N1, GPIO_NUM_12, GPIO_NUM_15 // UBlox NEO 6M RX, TX
#define GPS_INT GPIO_NUM_13 // 30ns accurary timepulse, to be external wired on pcb: NEO 6M Pin#3 -> GPIO13
```
Do I understand it correctly that I have to connect GPIO12 to RX and GPIO13 to TX? But what do I have connect to GPIO13? My module only offers VCC, GND, RX and TX pins. Is there a way to get it running or should I use another module which offers PPS as an additional pin (like https://www.aliexpress.com/item/4001136384325.html)?

**Comment by cyberman54** [comment]
You need PPS line only for high precision time of day interrupt (GPS_INT). 

The GPS will work with RX/TX only.

If you don't need or have PPS, leave GPS_INT disabled.

**Comment by marsaiz** [comment]
Hola, te cuento que hice para que me funcione. Uso una placa Lilyo T3_v1.6.1 con un gps NEO 6M v2. el archivo ttgov21new.h queda con estas líneas descomentadas: 
// GPS settings
#define HAS_GPS 1 // use on board GPS
#define GPS_SERIAL 9600, SERIAL_8N1, GPIO_NUM_12, GPIO_NUM_15 // Configurar NEO 6M: RX en GPIO 12, TX en GPIO 15
#define GPS_INT GPIO_NUM_13 // 30ns accurary timepulse, to be external wired on pcb: NEO 6M Pin#3 -> GPIO13// GPS settings
Y los pines en la placa son RX 12 y TX 15, porsupuesto tienes que conectarlos inversamente.
Saludos
